bevm/service.go

```diff
@@ -36,6 +36,41 @@ func init() {
 	log.ErrFatal(err)
 }
 
+// Retrieve the StateTrie, waiting some time in case the minimum block index
+// required has not yet been reached.
+func getReadOnlyStateTrie(bcService *byzcoin.Service, byzcoinID []byte,
+	minBlockIndex int) (rst byzcoin.ReadOnlyStateTrie, err error) {
+	const maxTries = 3
+
+	for triesLeft := maxTries; ; {
+		rst, err = bcService.GetReadOnlyStateTrie(byzcoinID)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to retrieve ReadOnlyStateTrie: "+
+				"%v", err)
+		}
+
+		currentIndex := rst.GetIndex()
+
+		log.Lvlf2("[%v] rst index: %v >= %v?", maxTries-triesLeft,
+			currentIndex, minBlockIndex)
+
+		if currentIndex >= minBlockIndex {
+			break
+		}
+
+		triesLeft--
```

Comment:
Why is this not in the `for` loop above? Then you could return on success in line 58, and return the error if it goes out of the for-loop.

Reply:
I just found it clearer this way.